test/OrchardCore.Tests/Shell/ShellHostTests.cs

```diff
@@ -0,0 +1,80 @@
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Builders;
+using OrchardCore.Environment.Shell.Models;
+using OrchardCore.Tests.Apis.Context;
+using Xunit;
+
+namespace OrchardCore.Tests.Shell;
+
+public class ShellHostTests : SiteContext
+{
+    private readonly IShellHost _shellHost;
+
+    public ShellHostTests()
+    {
+        _shellHost = Site.Services.GetRequiredService<IShellHost>();
+    }
+
+    [Theory]
+    [InlineData("Test")]
+    [InlineData("orchardcore")]
+    [InlineData(ShellHelper.DefaultShellName)]
+
+    public async Task FindCaseInsensitiveShellSettings(string shellName)
+    {
+        await GetOrCreateShellContextAsync(shellName);
+        var anotherVariantOfShellName = GetDifferentVariant(shellName);
+
+        _shellHost.TryGetSettings(shellName, out var shellSettings);
+        _shellHost.TryGetSettings(anotherVariantOfShellName, out var anotherVariantOfShellSettings);
+
+        Assert.NotEqual(shellName, anotherVariantOfShellName);
+        Assert.NotNull(shellSettings);
+        Assert.NotNull(anotherVariantOfShellSettings);
+        Assert.Same(shellSettings, anotherVariantOfShellSettings);
+    }
+
+    [Theory]
+    [InlineData("Test")]
+    [InlineData("orchardcore")]
+    [InlineData(ShellHelper.DefaultShellName)]
+    public async Task FindCaseInsensitiveShellContext(string shellName)
+    {
+        await GetOrCreateShellContextAsync(shellName);
+        var anotherVariantOfShellName = GetDifferentVariant(shellName);
+
+        _shellHost.TryGetShellContext(shellName, out var shellContext);
+        _shellHost.TryGetShellContext(anotherVariantOfShellName, out var anotherVariantOfShellContext);
+
+        Assert.NotEqual(shellName, anotherVariantOfShellName);
+        Assert.NotNull(shellContext);
+        Assert.NotNull(anotherVariantOfShellContext);
+        Assert.Same(shellContext, anotherVariantOfShellContext);
+    }
+
+    private async Task<ShellContext> GetOrCreateShellContextAsync(string name)
+    {
+        var shellSettings = new ShellSettings()
+        {
+            Name = name,
+            State = TenantState.Uninitialized,
+        };
+
+        return await _shellHost.GetOrCreateShellContextAsync(shellSettings);
+    }
+
+    private static string GetDifferentVariant(string name)
```

Comment:
I'm not sure why we need this, you can supply various casing values in `Theory` args

Reply:
I thought this way we only provide one variable and the logic will compare it to another variant. This way we don’t have to provide to parameters. Is there a reason we should provide two parameters instead?

Reply:
Adding values with different cases will be more obvious to everyone instead of adding logic that doing something we don't know until we dig into the test